### ISSUE TITLE
Suggestion: Chieftain ARV with HESCO spawner for Patrol Bases

### ISSUE THREAD

**Issue by Hobbnob** [TheWreckingCrewUK/Insurgency-Redux-Legacy#34]
Thinking about how common Hesco barriers are around patrol bases, how difficult would it be to get Rik's ARV in with a function through ace interact to spawn/despawn a particular hesco barrier?

This is the kind of design outline I'm thinking:
The vehicle can spawn a long or short hesco wall, similar to the supply crate spawning the sandbag walls currently. The spawn would be delayed like with ace loading/unloading
Unlike the crate's function, the hesco would only spawn directly in front of the vehicle, with an error if the player tries to do anything dumb like spawning it on top of a vehicle.
The vehicle can also despawn a hesco barrier that is in front of it if the player chooses to deconstruct it.
Spawning a hesco wall uses supplies, and a lot of them.

We don't have the model (or really, the need) for a combat tractor and I think the ARV can work well as a repair/tow/engineering vehicle. Having the hesco placement limited to where the ARV can get would also stop hesco popping up in silly places, require teamwork to get it placed correctly and get hesco protection without resorting to anything immersion breaking like player placing the massive walls. The fact that it doesn't have any armament also means it needs teamwork to protect it while moving.

I had a quick go with the ARV and there are still quite a few large issues (namely crew cameras being inside meshes and the external textures lacking weathering) so I don't think this is something we can represent TWC with in the short term. If everyone likes this idea though I'll chat with Rik and see if he's willing to bump the obvious issues up in priority a bit.

**Comment by Bosenator** [comment]
Not part of the public mod set, and I don't believe he wishes it to be so - due to concerns.

**Comment by jayman9696** [comment]
Could we not just use the Base Arma 3 one? Although i'm not sure how much it would actually be used

**Comment by Hobbnob** [comment]
Yes and no, the problem with the Nemmera is that it has a 50 cal CROWS system with thermal. Even if we took the ammo out it's still got the optics.

**Comment by Hobbnob** [comment]
Doing a bit of research it seems the wolfhound fits the bill until we get something like a trojan in, given the wolfhound's lack of human storage capacity, lack of armament (we'd use the GPMG variant) and its IRL usage in fob building. The problem here is that I'm dumb and can't get stuff to reliably spawn in directly in front of me, let alone add ace interaction while in the vehicle and have it feed from supplies and cancel the spawn if it can't do it in the exact position without clipping into non-hesco objects. The ace explosives framework has a lot of benefits that would help here on the placement issue, but I haven't been successful in isolating the useful parts.

**Comment by Bosenator** [comment]
I don't think this would be significant work, but I will assign this for 1.1 for the moment. It would be nice to have more engineering type functionality within the framework, for mission caveats too.
